src/waypoint_cmd.cpp

```diff
@@ -292,6 +319,147 @@ CommandCost CmdBuildRailWaypoint(DoCommandFlag flags, TileIndex start_tile, Axis
+ * @param axis orientation (Axis)
+ * @param width width of waypoint
+ * @param height height of waypoint
+ * @param spec_class custom road stop class
+ * @param spec_index custom road stop id
+ * @param station_to_join station ID to join (NEW_STATION if build new one)
+ * @param adjacent allow waypoints directly adjacent to other waypoints.
+ * @return the cost of this operation or an error
+ */
+CommandCost CmdBuildRoadWaypoint(DoCommandFlag flags, TileIndex start_tile, Axis axis, uint8_t width, uint8_t height, RoadStopClassID spec_class, uint16_t spec_index, StationID station_to_join, bool adjacent)
+{
+	if (!IsValidAxis(axis)) return CMD_ERROR;
+	/* Check if the given station class is valid */
+	if (static_cast<uint>(spec_class) >= RoadStopClass::GetClassCount()) return CMD_ERROR;
+	const RoadStopClass *cls = RoadStopClass::Get(spec_class);
+	if (!IsWaypointClass(*cls)) return CMD_ERROR;
+	if (spec_index >= cls->GetSpecCount()) return CMD_ERROR;
+
+	const RoadStopSpec *roadstopspec = RoadStopClass::Get(spec_class)->GetSpec(spec_index);
+
+	/* The number of parts to build */
+	uint8_t count = axis == AXIS_X ? height : width;
+
+	if ((axis == AXIS_X ? width : height) != 1) return CMD_ERROR;
+	if (count == 0 || count > _settings_game.station.station_spread) return CMD_ERROR;
+
+	bool reuse = (station_to_join != NEW_STATION);
+	if (!reuse) station_to_join = INVALID_STATION;
+	bool distant_join = (station_to_join != INVALID_STATION);
+
+	if (distant_join && (!_settings_game.station.distant_join_stations || !Waypoint::IsValidID(station_to_join))) return CMD_ERROR;
+
+	TileArea roadstop_area(start_tile, width, height);
+
+	/* Total road stop cost. */
+	Money unit_cost;
+	if (roadstopspec != nullptr) {
+		unit_cost = roadstopspec->GetBuildCost(PR_BUILD_STATION_TRUCK);
+	} else {
+		unit_cost = _price[PR_BUILD_STATION_TRUCK];
+	}
+	StationID est = INVALID_STATION;
+	CommandCost cost = CalculateRoadStopCost(roadstop_area, flags, true, STATION_ROADWAYPOINT, axis, AxisToDiagDir(axis), &est, INVALID_ROADTYPE, unit_cost);
+	if (cost.Failed()) return cost;
+
+	Waypoint *wp = nullptr;
+	CommandCost ret = FindJoiningWaypoint(est, station_to_join, adjacent, roadstop_area, &wp, true);
+	if (ret.Failed()) return ret;
+
+	/* Check if there is an already existing, deleted, waypoint close to us that we can reuse. */
+	TileIndex center_tile = start_tile + (count / 2) * TileOffsByDiagDir(AxisToDiagDir(OtherAxis(axis)));;
+	if (wp == nullptr && reuse) wp = FindDeletedWaypointCloseTo(center_tile, STR_SV_STNAME_WAYPOINT, _current_company, true);
+
+	if (wp != nullptr) {
+		/* Reuse an existing waypoint. */
+		if (!HasBit(wp->waypoint_flags, WPF_ROAD)) return CMD_ERROR;
+		if (wp->owner != _current_company) return_cmd_error(STR_ERROR_TOO_CLOSE_TO_ANOTHER_WAYPOINT);
+
+		ret = wp->rect.BeforeAddRect(start_tile, width, height, StationRect::ADD_TEST);
+		if (ret.Failed()) return ret;
+	} else {
+		/* allocate and initialize new waypoint */
```

Comment:
This comment isn't accurate, maybe:
```suggestion
		/* Check if we can create a new waypoint. */
```

Reply:
Where possible, things have been done in an identical way to rail waypoints, I can change this however.